tinytodo/src/context.rs

```diff
@@ -288,24 +282,83 @@ impl AppContext {
 
     #[tracing::instrument(skip(policy_set))]
     fn update_policy_set(&mut self, policy_set: PolicySet) -> Result<AppResponse> {
-        self.policies = policy_set;
+        self.policies.set_static_store(policy_set)?;
         info!("Reloaded policy set");
         Ok(AppResponse::Unit(()))
     }
 
     fn add_share(&mut self, r: AddShare) -> Result<AppResponse> {
         self.is_authorized(&r.uid, &*ACTION_EDIT_SHARE, &r.list)?;
         let list = self.entities.get_list(&r.list)?;
-        let team_uid = list.get_team(r.role).clone();
+        let kind = r.share_kind();
+        let team_uid = list.get_team(kind).clone();
         let target_entity = self.entities.get_user_or_team_mut(&r.share_with)?;
         target_entity.insert_parent(team_uid);
+
+        if kind == ShareKind::Timebox {
+            self.update_timebox(
+                r.share_with,
+                r.list,
+                r.duration_in_seconds.map(Duration::from_secs),
+            )?;
+        }
+
         Ok(AppResponse::Unit(()))
     }
 
+    #[tracing::instrument]
```

Comment:
All of the functionality to manage and update timeboxes and such doesn't feel like it belongs in `context.rs`, which ought to just have the handlers for the commands. Can you move it somewhere else, e.g., to `util.rs` ?

Reply:
Actually, I'd make a new file, `timebox.rs` and put all the timebox related stuff there, other than the outermost handler.